godtoolsTests/App/Share/Common/StringSearcher/StringSearcherTests.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  StringSearcherTests.swift
+//  godtoolsTests
+//
+//  Created by Rachael Skeath on 7/21/25.
+//  Copyright © 2025 Cru. All rights reserved.
+//
+
+import Testing
+@testable import godtools
+import Combine
+
+struct StringSearcherTests {
+    
+    struct TestArgument {
+        let searchTerm: String
+        let expectedSearchResults: [MockStringSearchable]
+    }
+    
+    struct MockStringSearchable: StringSearchable, Equatable {
+        let searchableStrings: [String]
+    }
+    
+    private static let englishSearchable = MockStringSearchable(searchableStrings: ["English", "English"])
+    private static let albanianSearchable = MockStringSearchable(searchableStrings: ["Albanian", "shqip"])
+    private static let croatianSearchable = MockStringSearchable(searchableStrings: ["Croatian", "hrvatski"])
+    private static let englishBahrainSearchable = MockStringSearchable(searchableStrings: ["English (Bahrain)", "English (Bahrain)"])
+    private static let frenchSearchable = MockStringSearchable(searchableStrings: ["French", "français"])
+    private static let spanishSearchable = MockStringSearchable(searchableStrings: ["Spanish", "español"])
+    
+    @Test(
+        "Search a list of StringSearchables with a search term string and return case-insensitive matches",
+        arguments: [
+            TestArgument(searchTerm: "English", expectedSearchResults: [englishSearchable, englishBahrainSearchable]),
+            TestArgument(searchTerm: "Bangla", expectedSearchResults: []),
+            TestArgument(searchTerm: "SH", expectedSearchResults: [englishSearchable, albanianSearchable, englishBahrainSearchable, spanishSearchable]),
+            TestArgument(searchTerm: "french", expectedSearchResults: [frenchSearchable]),
+            TestArgument(searchTerm: "ESPAÑOL", expectedSearchResults: [spanishSearchable]),
+            TestArgument(searchTerm: "ai", expectedSearchResults: [englishBahrainSearchable, frenchSearchable]),
+            TestArgument(searchTerm: "", expectedSearchResults: [englishSearchable, albanianSearchable, croatianSearchable, englishBahrainSearchable, frenchSearchable, spanishSearchable]),
```

Comment:
Hey @rachaelblue another thing I like to do is organize my tests by functionality or intent.  

For example, because the empty search string isn't part of the case sensitivity logic or main search logic, instead it returns immediately with the searchables, I'd create another test method with something like "empty search string should return with original list of searchables". 

Doing this high-lights each piece of functionality or intent of the StringSearcher.

Reply:
This is a good idea, I was thinking having all the test arguments lumped together could be hard to follow.